front/components/tables/TablePicker.tsx

```diff
@@ -209,28 +212,26 @@ export default function TablePicker({
                         </div>
                       </div>
                     ))}
-                  {allTablesMap.size === 0 && (
-                    <span className="block px-4 pt-2 text-sm text-gray-700">
-                      No tables found
-                    </span>
-                  )}
+                  {debouncedSearch && allTablesMap.size === 0 &&
+                    !showTableLoaders && (
+                      <span className="block px-2 mt-2 text-sm text-gray-700">
+                        No tables found
+                      </span>
+                    )}
                 </div>
                 <InfiniteScroll
                   nextPage={() => {
                     handleLoadNext(nextPageCursor);
                   }}
                   hasMore={!!nextPageCursor}
-                  isValidating={isTablesLoading || isDebouncing}
-                  isLoading={isTablesLoading || isDebouncing}
-                >
-                  {(isTablesLoading || isDebouncing) && !allTablesMap.size && (
-                    <div className="py-2 text-center text-sm text-element-700">
-                      Loading tables...
+                  showLoader={showTableLoaders}
```

Comment:
I couldn't resist and did a few design improvements too here, and it should now look like this. 

Main changes are:
- loading message with animation
- improve spacing
- not showing "no tables found" when there is no search term

I could do more but I needed to stop otherwise I'm going another rabbit hole 😄


https://github.com/user-attachments/assets/cc36e410-b382-43ff-8f14-c28cdf99a282

Reply:
Very nice 💯